src/diffpy/labpdfproc/tools.py

```diff
@@ -213,3 +213,36 @@ def load_package_info(args):
     metadata = get_package_info("diffpy.labpdfproc")
     setattr(args, "package_info", metadata["package_info"])
     return args
+
+
+def _set_anode_type(args):
```

Comment:
this may be doing too much.  i.e., "magic"?

I think the ucs are
uc1
1. user specifies a standard anode material
2. labpdfproc looks up the wavelength
3. lpp stores the wavelength and anode type in metadata

UC2
1. user specifies a (numeric) wavelength
2. lpp stores the (numeric) wavelength in metadata.

I don't thing there is a reverse UC where a numeric value stored and lpp saves the anode type.

Reply:
I think we only took care of wavelength in the current code. anode type will be Mo if user doesn't specify it. So an input of wavelength=1.54 will produce anode_type=Mo in the current code. Shall I modify this in the wavelength function, or should I add a new function that take care of anode type?

Reply:
I am not sure what you are saying here.    I want to remove the magic: "input of wavelength=1.54 will produce anode_type=Mo"

We don't want to guess where the user got the 1.54 A data from, maybe from a synchrotron?

Reply:
I mean we might need a separate function (or I can modify the existing wavelength function?) to remove the magic because in the previous issues we only checked that args has the correct wavelength (i.e. we didn't remove the anode type for UC2, but anode type is loaded automatically into the args because it's one of the arguments).
I got your point on not doing the reverse case.